lib/blocks.php

```diff
@@ -375,3 +375,21 @@ function gutenberg_register_theme_block_category( $categories ) {
 }
 
 add_filter( 'block_categories', 'gutenberg_register_theme_block_category' );
+
+/**
+ * Checks whether the current block type supports the feature requested.
+ *
+ * @param WP_Block_Type $block_type Block type to check for support.
+ * @param string        $feature    Name of the feature to check support for.
+ * @param mixed         $default    Fallback value for feature support, defaults to false.
+ *
+ * @return boolean                  Whether or not the feature is supported.
+ */
+function gutenberg_block_has_support( $block_type, $feature, $default = false ) {
+	$block_support = $default;
+	if ( property_exists( $block_type, 'supports' ) ) {
+		$block_support = _wp_array_get( $block_type->supports, $feature, $default );
+	}
+
+	return true === $block_support || is_array( $block_support );
```

Comment:
Should we use this function in other places like [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/colors.php#L15) and in other block-supports?

Reply:
Probably, though the color is a bit granular, so I'm not sure if it applies properly there, I'll check.

Reply:
I've used everywhere except colors and typography where it's a bit more advanced.